herbert_core/horace_paths.m

```diff
@@ -1,58 +1,64 @@
 classdef horace_paths
+    % Helper class, containing information abut main path-es used by Horace    
+    % Its main properties value return the following path-es:
+    %
+    % herbert          - path to main Herbert code
+    % horace           - path to main Horace code
+    % root             - path to folder, which contains all Horace code
+    % test_common      - path to test data used by multiple tests
+    % test_common_func - path to functions used by multiple tests
+    % admin            - path to folder, containing main srcipts to control
+    %                    Horace& Herbert code, mainly used in installation and
+    %                    building Horace
+    % test             - path to folder with all init tests
+    % low_level        - path to folder with C++ code
+    % bm               - path to benchmarking code
+    % bm_common        - path to benchmarking data
+    % bm_common_func   - math to common function used by all benchmarging tests
 
     properties(Dependent)
-        herbert
-        horace
-        root
-        test_common
-        test_common_func
-        admin
-        test
-        low_level
-        bm
-        bm_common
-        bm_common_func
+        herbert  % path to main Herbert code
+        horace   % path to main Horace code
+        root     % path to folder, which contains all Horace code
+        test_common % path to test data used by multiple tests
+        test_common_func % path to functions used by multiple tests
+        admin    % path to folder, containing main srcipts to control Horace& Herbert code
+        test     % path to folder with all init tests
+        low_level % path to folder with C++ code
+        bm        % path to benchmarking code
+        bm_common % path to benchmarking data
+        bm_common_func % math to common function used by all benchmarging tests
+    end
+    properties(Access=protected)
+        herbert_
+        horace_
+        root_
     end
 
-    methods
-
-        function herbert_path = get.herbert(obj)
-            global herbert_path
-            if ~exist('herbert_path', 'var') || isempty(herbert_path)
 
-                herbert_path = obj.get_folder('herbert_init');
-            end
+    methods
+        function her_ = get.herbert(obj)
+            her_= obj.herbert_;
         end
-
-        function horace_path = get.horace(obj)
-            global horace_path
-            if ~exist('horace_path', 'var') || isempty(horace_path)
-
-                horace_path = obj.get_folder('horace_init');
-            end
+        function hor_ = get.horace(obj)
+            hor_= obj.horace_;
         end
-
-        function root_path = get.root(obj)
-            global root_path
-            if ~exist('root_path', 'var') || isempty(root_path)
-                root_path = fileparts(obj.get_folder('horace_init'));
-            end
+        function rootp = get.root(obj)
```

Comment:
Change required: See earlier comment about variable names, and change the name here accordingly.